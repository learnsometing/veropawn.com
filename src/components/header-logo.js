/*
*
* Header Logo
*
* A link to the home page that contains the site's SVG logo.
*
*/

import React from "react";
import headerStyles from "./header.module.css";
import { useStaticQuery, graphql } from "gatsby";

import { Link } from "gatsby";

export const PureHeaderLogo = ({ data }) => (
  <Link
    to='/'
    className={headerStyles.link}
  >
    <span className={headerStyles.iconWrapper}>
      <div className={headerStyles.svgWrapper}>
        <img
          className={headerStyles.svg}
          src={data.logo.publicURL}
          alt="Cash Pawn and Jewelry Logo"
        />
      </div>
    </span>
  </Link>
)

export default () => {
  const data = useStaticQuery(graphql`
    query LogoQuery {
      logo: file(relativePath: { eq: "logo.svg" }) {
        publicURL
      }
    }
  `);

  return (
    <PureHeaderLogo data={data}></PureHeaderLogo>
  );
}