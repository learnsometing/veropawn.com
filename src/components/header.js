import PropTypes from "prop-types"
import React from "react"
import { useStaticQuery, graphql } from "gatsby"

import MainMenu from "./main-menu"
import HeaderLogo from "./header-logo"
import headerStyles from "./header.module.css"

const Header = () => {
  const data = useStaticQuery(graphql`
    query LogoQuery {
      logo: file(relativePath: { eq: "logo.svg" }) {
        publicURL
      }
    }
  `)

  return (
    <header className={headerStyles.header}>
      <nav className={headerStyles.nav}>
        <HeaderLogo logo={data.logo} />
        <MainMenu />
      </nav >
    </header >
  )
}

// const menuItems = useStaticQuery(graphql`
//   query Q {
//     allInvJson(
//       sort:{
//         fields:[category]
//         order:ASC
//       }
//     ){
//       edges{
//         node{
//           category
//           subcategory
//           slug
//         }
//       }
//     }
//   }
// `)
// const edges = menuItems.allInvJson.edges;
// const nodes = edges.map(edge => {
//   return edge.node;
// })
// const info = nodes.map(node => {
//   return { category: node.category, subcategory: node.subcategory, slug: node.slug }
// })

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
