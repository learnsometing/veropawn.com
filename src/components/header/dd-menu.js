import React from "react";
import { Link } from "gatsby";
import { FaAngleUp, FaAngleDown } from "react-icons/fa";
import PropTypes from "prop-types";

import ddMenu from "./dd-menu.module.css";
import layout from "../../styles/layout.module.css";

export const DDMenuBtn = ({ children, isIconAfterText, isNavButton, onClick, text }) => {
  let buttonClass = isNavButton ? ddMenu.navButton : ddMenu.button;
  buttonClass += ` ${layout.rowCenterCenter}`;

  let childNodes = isIconAfterText
    ? <>
      {text}
      {children}
    </>
    : <>
      {children}
      {text}
    </>

  return (
    <li className={layout.rowCenterCenter}>
      <button className={buttonClass} onClick={onClick}>
        {childNodes}
      </button>
    </li>
  );
};

DDMenuBtn.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object
  ]),
  isIconAfterText: PropTypes.bool,
  isNavButton: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string
};

export const DDMenuHeader = (props) => {
  const headerClass = `${layout.rowCenterCenter} ${ddMenu.heading}`
  return (
    <li className={ddMenu.listItem}>
      <header className={headerClass} data-testid="dd-menu-heading">
        <h1 style={{ fontSize: '1.375rem' }}>
          {props.children}
        </h1>
      </header>
    </li>
  );
}

export const DDMenuLink = (props) => {
  const linkClass = `${layout.rowCenterCenter} ${ddMenu.link}`
  return (
    <li className={ddMenu.listItem}>
      <Link className={linkClass} to={props.link}>
        {props.children}
        {props.text}
      </Link>
    </li>
  );
};

const DDMenuStatusIcon = ({ isOpen }) => {
  let icon = <FaAngleDown data-testid={"fa-angle-down-icon"} />;

  if (isOpen) {
    icon = <FaAngleUp data-testid={"fa-angle-up-icon"} />;
  }

  return icon;
}

export const DDMenuToggleBtn = ({ isOpen, value, toggleMenu }) => {
  /*
  * Accepts a dropdown toggle function and text as props.
  *
  * Switches between up and down arrow icons when clicked.
  */

  const className = `${layout.rowCenterCenter} ${ddMenu.toggleMenuBtn}`;

  return (
    <button
      className={className}
      onClick={toggleMenu}
    >
      {value} <DDMenuStatusIcon isOpen={isOpen} />
    </button>
  );
}