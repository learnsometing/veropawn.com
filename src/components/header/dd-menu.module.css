$bottom-border: 1px solid rgba(64, 61, 52, 0.6);

.heading, .link, .button, .navButton {
  width: 100%;
  padding: .5rem 0;
  background-color: #e7e8c8;
  border-bottom: $bottom-border;
  color: #403D34;
  font-weight: 500;
  z-index: 1;
}

.heading {
  padding: .75rem;
  color: #36733F;
  text-transform: uppercase;
}

.navButton {
  font-size: 1.125rem;
}

.button:hover, .link:hover, .toggleMenuBtn:hover{
  cursor: pointer;
  text-decoration: underline;
}

.toggleMenuBtn {
  color: #43443f;
  font-weight: bold;
}

@media only screen and (min-width: 667px) {
  .toggleMenuBtn {
    height: 85%;
  }
}