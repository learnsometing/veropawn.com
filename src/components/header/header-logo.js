/*
*
* Header Logo
*
* A link to the home page that contains the site's SVG logo.
*
*/

import React from "react";
import headerStyles from "./header.module.css";

import { Link } from "gatsby";

export default ({ logo }) => (
  <Link to='/' className={headerStyles.link}>
    <span className={headerStyles.iconWrapper}>
      <div className={headerStyles.svgWrapper}>
        <img
          className={headerStyles.svg}
          src={logo.publicURL}
          alt="Cash Pawn and Jewelry Logo"
        />
      </div>
    </span>
  </Link>
)