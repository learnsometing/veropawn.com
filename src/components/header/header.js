// Node_Modules Imports
import React, { useState } from "react";
import { graphql, Link, useStaticQuery } from "gatsby";
import sizeMe from "react-sizeme";

// Internal Imports
import headerStyles from "../header/header.module.css";
import HeaderModalLeft from "../modals/header-modal-left";
import CollapsedMainMenu from "../header/collapsed-main-menu";
import ExpandedMainMenu from "../header/expanded-main-menu";

export const HeaderLogo = ({ logo }) => (
  // A link to the home page that contains the site's SVG logo

  <Link to='/' className={headerStyles.link}>
    <span className={headerStyles.iconWrapper}>
      <div className={headerStyles.svgWrapper}>
        <img
          className={headerStyles.svg}
          src={logo.publicURL}
          alt="Cash Pawn and Jewelry Logo"
        />
      </div>
    </span>
  </Link>
);

export const withHeaderModalLeft = MenuComponent => {
  return props => {
    const [isOpen, setIsOpen] = useState(false);
    const [modalMenu, setModalMenu] = useState(null);

    const toggleMenu = (menu) => {
      setIsOpen(!isOpen);
      setModalMenu(modalMenu ? null : menu);
    };

    const closeModal = () => {
      setIsOpen(false);
      setModalMenu(null);
    };

    return (
      <>
        <MenuComponent
          allInvJson={props.allInvJson}
          allMarkdownRemark={props.allMarkdownRemark}
          isOpen={isOpen}
          toggleMenu={toggleMenu}
        />
        <HeaderModalLeft isOpen={isOpen} closeModal={closeModal} >
          {modalMenu}
        </HeaderModalLeft>
      </>
    );
  }
}

export const PureHeader = ({ size, ...props }) => {
  let HeaderMenu;

  if (size.width < 668) {
    HeaderMenu = withHeaderModalLeft(CollapsedMainMenu);
  } else {
    HeaderMenu = withHeaderModalLeft(ExpandedMainMenu);
  }

  return (
    <div className={headerStyles.headerWrapper}>
      <header className={headerStyles.header}>
        <nav className={headerStyles.nav}>
          <HeaderLogo logo={props.logo} />
          <HeaderMenu
            allInvJson={props.allInvJson}
            allMarkdownRemark={props.allMarkdownRemark}
          />
          <span>{size.width}</span>
        </nav >
      </header >
    </div >
  );
}

export const Header = ({ size }) => {
  const { logo, allMarkdownRemark, allInvJson } = useStaticQuery(
    graphql`
      query {
        logo: file(relativePath: { eq: "logo.svg" }) {
          publicURL
        }
        
        allMarkdownRemark(sort: {fields: frontmatter___title}){
          nodes{
            id
            frontmatter{
              title
            }
            fields {
              slug
            }
          }
        }

        allInvJson(sort: {fields: category}) {
          distinct(field: category)
          nodes{
            category
            id
            subcategory
            slug
          }
        }
      }
    `
  );

  return (
    <PureHeader
      size={size}
      logo={logo}
      allMarkdownRemark={allMarkdownRemark}
      allInvJson={allInvJson}
    />
  );
}

export default sizeMe()(Header);