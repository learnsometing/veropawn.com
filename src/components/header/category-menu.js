// node_modules
import React, { useState } from "react";
import PropTypes from "prop-types";
import { FaAngleLeft } from "react-icons/fa";

// components and functions
import { DDMenuBtn, DDMenuLink, DDMenuHeader } from "../dropdown-menu/dd-menu";
import { prettifyCatOrSubcatName } from "../util/text-formatting";

export const sortByButtonText = (ComponentA, ComponentB) => {
  if (typeof ComponentA.props.text !== 'string' || typeof ComponentB.props.text !== 'string') {
    throw new TypeError('Invalid prop `text` supplied to `DDMenuBtn`, expected string');
  }

  if (ComponentA.props.text < ComponentB.props.text) {
    return -1;
  } else if (ComponentA.props.text > ComponentB.props.text) {
    return 1;
  }
  return 0;
}

const CategoryMenuBtns = ({ data, onClick }) => {
  /*
  * Returns a list of buttons that each have a category name as text.
  * 
  * When clicked, each button should render a list of links to the category's
  * subcategory pages.
  */

  const filterNodesByCategory = (category) => {
    return data.nodes.filter(node => node.category === category);
  }

  return (
    data.distinct.map(category => {
      let prettyCategory = prettifyCatOrSubcatName(category);

      return (
        <DDMenuBtn
          key={category}
          onClick={onClick.bind(null, prettyCategory, filterNodesByCategory(category))}
          text={prettyCategory}
        />
      );
    }).sort((CompA, CompB) => sortByButtonText(CompA, CompB))
  );
}

CategoryMenuBtns.propTypes = {
  data: PropTypes.object,
  onClick: PropTypes.func
};

const SubcategoryMenuLinks = ({ nodes }) => {
  return (
    nodes.map(node => {
      const prettySubcategory = prettifyCatOrSubcatName(node.subcategory);
      return (
        <DDMenuLink
          key={node.id}
          link={node.fields.slug}
          text={prettySubcategory}
        />
      );
    }).sort((CompA, CompB) => sortByButtonText(CompA, CompB))
  );
}

SubcategoryMenuLinks.propTypes = {
  nodes: PropTypes.array
};

export default ({ data, ...props }) => {
  /*
  * Category Menu
  *
  * A dropdown menu that exposes the categories of items in the pawn shop's
  * inventory to the user.
  */

  const [subcatMenuOpen, setSubcatMenuOpen] = useState(false);
  const [subcatMenuHeader, setSubcatMenuHeader] = useState('');
  const [subcatMenuLinks, setSubcatMenuLinks] = useState([]);

  const openSubcatMenu = (header, links) => {
    setSubcatMenuOpen(true);
    setSubcatMenuHeader(header);
    setSubcatMenuLinks(links);
  }

  const closeSubcatMenu = () => {
    setSubcatMenuOpen(false);
    setSubcatMenuHeader('');
    setSubcatMenuLinks([]);
  }

  return (
    subcatMenuOpen
      ?
      <>
        <DDMenuBtn key="back-to-categories" onClick={closeSubcatMenu}>
          <FaAngleLeft data-testid="angle-left-icon" />
          {"Categories"}
        </DDMenuBtn>
        <DDMenuHeader key={subcatMenuHeader}>
          {subcatMenuHeader}
        </DDMenuHeader>
        <SubcategoryMenuLinks nodes={subcatMenuLinks} />
      </>
      :
      <>
        <DDMenuHeader key={"categories"}>
          Categories
        </DDMenuHeader>
        <CategoryMenuBtns data={data} onClick={openSubcatMenu} />
      </>
  );
}