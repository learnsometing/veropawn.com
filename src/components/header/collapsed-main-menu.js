/*
* Collapsed Main Menu
*
* The main menu that is rendered when the screen width is below 668 px.
*/

// Node_Modules Imports
import React, { useState } from "react";
import { FaAngleLeft, FaAngleRight } from "react-icons/fa";

// Internal Imports
import { DDMenuHeader, DDMenuBtn, DDMenuToggleBtn } from "./dd-menu";
import MDPageLinks from "./md-page-links";

import CategoryMenu from "./category-menu";

export const MainMenu = (props) => {
  const [categoryMenuOpen, setCategoryMenuOpen] = useState(false);

  const openCategoryMenu = () => {
    setCategoryMenuOpen(true);
  }

  const backToMainMenu = () => {
    setCategoryMenuOpen(false);
  }

  return (
    categoryMenuOpen
      ? <>
        <DDMenuBtn
          key="back-to-main-menu"
          onClick={backToMainMenu}
          text={"Main Menu"}
          isNavButton={true}
        >
          <FaAngleLeft data-testid="fa-angle-left-icon" />
        </DDMenuBtn>
        <CategoryMenu data={props.allPagesJson} />
      </>
      : <>
        <DDMenuHeader key="main-menu">
          Main Menu
        </DDMenuHeader>
        <DDMenuBtn
          key="browse-by-category"
          onClick={openCategoryMenu}
          text={"Categories"}
          isNavButton={true}
          isIconAfterText={true}
        >
          <FaAngleRight data-testid="fa-angle-right-icon" />
        </DDMenuBtn>
        <MDPageLinks data={props.allMarkdownRemark} collapsed={true} />
      </>
  );
}

export default (props) => {
  const toggleMenu = props.toggleMenu.bind(
    null,
    <MainMenu
      allPagesJson={props.allPagesJson}
      allMarkdownRemark={props.allMarkdownRemark}
    />
  );

  return (
    <DDMenuToggleBtn
      isOpen={props.isOpen}
      value={"Menu"}
      toggleMenu={toggleMenu}
    />
  );
}