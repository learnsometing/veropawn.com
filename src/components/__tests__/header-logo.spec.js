/*
* Tests for the HeaderLogo component.
*/

import React from "react";
import renderer from "react-test-renderer";

import logoData from "../__fixtures__/logo-data"
import { PureHeaderLogo as HeaderLogo } from "../header/header-logo";

describe("HeaderLogo component", () => {
  test("Matches the snapshot", () => {
    const { data } = logoData
    const headerLogo = renderer.create(<HeaderLogo data={data} />);
    expect(headerLogo.toJSON()).toMatchSnapshot();
  });
});