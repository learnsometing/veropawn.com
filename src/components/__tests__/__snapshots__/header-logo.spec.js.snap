// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogo component Matches the snapshot 1`] = `
<a
  className="link"
  href="/"
>
  <span
    className="iconWrapper"
  >
    <div
      className="svgWrapper"
    >
      <img
        alt="Cash Pawn and Jewelry Logo"
        className="svg"
        src="/static/logo-c88db6fb705fc71a30bcbcbbdd1c72b8.svg"
      />
    </div>
  </span>
</a>
`;
